src/main/java/me/ryanhamshire/GriefPrevention/Claim.java

```diff
@@ -443,214 +395,184 @@ public boolean hasExplicitPermission(Player player, ClaimPermission level)
+            if (node.length() < 3 || node.charAt(0) != '[' || node.charAt(node.length() - 1) != ']') continue;
 
             // Check if level is high enough and player has node
             if (level.isGrantedBy(stringToPermission.getValue())
                     && player.hasPermission(node.substring(1, node.length() - 1)))
-            {
                 return true;
-            }
         }
 
         return false;
     }
 
-    //break permission check
-    public String allowBreak(Player player, Material material)
-    {
-        //if under siege, some blocks will be breakable
-        if (this.siegeData != null || this.doorsOpen)
-        {
-            boolean breakable = false;
+    /**
+     * Check whether or not a player has a certain level of trust for an action.
+     *
+     * @param player the Player performing the action
+     * @param permission the ClaimPermission required to perform the action
+     * @param event the Event resulting in the permission check or null if no event was fired
+     * @return the denial message, or null if the action is allowed
+     */
+    public String checkPermission(Player player, ClaimPermission permission, Event event) {
+        ClaimPermissionCheckEvent permissionEvent = new ClaimPermissionCheckEvent(player, this, permission, event);
 
-            //search for block type in list of breakable blocks
-            for (int i = 0; i < GriefPrevention.instance.config_siege_blocks.size(); i++)
-            {
-                Material breakableMaterial = GriefPrevention.instance.config_siege_blocks.get(i);
-                if (breakableMaterial == material)
-                {
-                    breakable = true;
-                    break;
-                }
-            }
+        // Set denial message (if any) using default behavior, then allow addons to modify.
+        permissionEvent.setDenialMessage(getPermissionMessage(player, permission, event));
 
-            //custom error messages for siege mode
-            if (!breakable)
-            {
-                return GriefPrevention.instance.dataStore.getMessage(Messages.NonSiegeMaterial);
-            }
-            else if (player.getUniqueId().equals(this.ownerID))
-            {
-                return GriefPrevention.instance.dataStore.getMessage(Messages.NoOwnerBuildUnderSiege);
-            }
-            else
-            {
-                return null;
-            }
-        }
+        Bukkit.getPluginManager().callEvent(permissionEvent);
 
-        //if not under siege, build rules apply
-        return this.allowBuild(player, material);
+        return permissionEvent.getDenialMessage();
     }
 
-    //access permission check
-    public String allowAccess(Player player)
-    {
-        //following a siege where the defender lost, the claim will allow everyone access for a time
-        if (this.doorsOpen) return null;
+    private String getPermissionMessage(Player player, ClaimPermission permission, Event event) {
```

Comment:
This is a helper method for `getPermission` used internally to not fire subsequent events when checking claim parents if necessary. I was torn on the name of this method.